### PR TITLE
[WiP][Testing] Add new parameters to .ini file to read them by SDL

### PR DESCRIPTION
**Pull request by dcherniev**
Added new parameters 'menuIcon' and 'menuTitle' to the .ini file.
Added to SDL ability to read default values for those parameters from ini file.
Also added to SDL resetting of those parameters in ResetGlobalProperties request.

Reopened from PR: #931 (Moved feature from branch "feature/pre_5.0" to branch "develop")

### CODE REVIEW

**Comment by dcherniev** [comment]
Please review : @vlantonov, @Kozoriz, @LuxoftAKutsan, @anosach-luxoft, @dtrunov

File: src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc [hunk 1]
```diff
@@ -0,0 +1,461 @@
...
+        .WillByDefault(
+            Return(am::HmiInterfaces::HMI_INTERFACE_BasicCommunication));
+    ON_CALL(mock_hmi_interfaces_, GetInterfaceState(_))
+        .WillByDefault(Return(am::HmiInterfaces::STATE_AVAILABLE));
+
+    Mock::VerifyAndClearExpectations(&mock_message_helper_);
+  }
+
+  void TearDown() OVERRIDE {
+    Mock::VerifyAndClearExpectations(&mock_message_helper_);
```
Comment (vlantonov) [inline src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc]: @dcherniev According to the [Google Test FAQ](https://github.com/google/googletest/blob/master/googletest/docs/FAQ.md#should-i-use-the-constructordestructor-of-the-test-fixture-or-the-set-uptear-down-function) `SetUp` and `TearDown` are to be used when there is a chance an exception to be thrown. This is not the case with `Mock::VerifyAndClearExpectations(&mock_message_helper_)` so it should be moved to destructor. Also as it is a static object there is a chance some other Unit test to influence it, so additionally `Mock::VerifyAndClearExpectations(&mock_message_helper_);` should be put in constructor.

File: src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc [hunk 2]
```diff
@@ -0,0 +1,461 @@
...
+  std::vector<std::string> time_out_prompt;
+  time_out_prompt.push_back("time_out");
+  EXPECT_CALL(app_mngr_settings_, time_out_promt())
+      .WillOnce(ReturnRef(time_out_prompt));
+
+  EXPECT_CALL(*mock_app_, set_timeout_prompt(_));
+
+  smart_objects::SmartObjectSPtr prompt =
+      utils::MakeShared<smart_objects::SmartObject>();
+  *prompt = "prompt";
```
Comment (dcherniev) [inline src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc]: @vlantonov Maybe I should better use the SmartType_Map like in other instances of this promp setup:
const smart_objects::SmartObjectSPtr so_time_out_prompt =
      utils::MakeShared<smart_objects::SmartObject>(
          smart_objects::SmartType_Map);

File: src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc (hunk 1 above)
Comment (dcherniev) [inline src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc]: @vlantonov Fixed, please check 70853c0

File: src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc [hunk 3]
```diff
@@ -0,0 +1,461 @@
...
+  MessageSharedPtr command_result;
+  EXPECT_CALL(
+      app_mngr_,
+      ManageMobileCommand(_, am::commands::Command::CommandOrigin::ORIGIN_SDL))
+      .WillOnce(DoAll(SaveArg<0>(&command_result), Return(true)));
+
+  command_->Run();
+  EXPECT_EQ(
+      (*command_result)[am::strings::msg_params][am::strings::success].asBool(),
+      false);
```
Comment (dcherniev) [inline src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc]: @vlantonov Fixed, please check 70853c0

File: src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc [hunk 4]
```diff
@@ -0,0 +1,461 @@
...
+  (*msg_)[am::strings::params][am::hmi_response::code] =
+      hmi_apis::Common_Result::eType::UNSUPPORTED_RESOURCE;
+
+  (*msg_)[am::strings::msg_params][am::strings::properties][0] =
+      mobile_apis::GlobalProperty::TIMEOUTPROMPT;
+  (*msg_)[am::strings::msg_params][am::strings::properties][1] =
+      mobile_apis::GlobalProperty::MENUNAME;
+
+  std::vector<std::string> time_out_prompt;
+  time_out_prompt.push_back("time_out");
```
Comment (dcherniev) [inline src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc]: @vlantonov Fixed, please check 70853c0

File: src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc (hunk 2 above)
Comment (dcherniev) [inline src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc]: @vlantonov Fixed, please check 70853c0

File: src/appMain/smartDeviceLink.ini [hunk 5]
```diff
@@ -211,6 +211,10 @@ ApplicationListUpdateTimeout = 2000
 ThreadPoolSize = 1
 ; The max size of hash which is used by OnHashUpdated
 HashStringSize = 32
+; Menu title default text
+MenuTitle = MENU
+; Path to default menu icon
```
Comment (LuxoftAKutsan) [inline src/appMain/smartDeviceLink.ini]: @dcherniev is it rely should be empty? If yes, please specify in comment where it rely will refers to by default

**Comment by LuxoftAKutsan** [comment]
@dcherniev what status of this PR?

**Comment by dcherniev** [comment]
@LuxoftAKutsan Well previosly (after my changes was approved) I have sent a letter to test team "Request for testing CRQ [APPLINK-28164]". However I didn't updated comment for "MenuIcon" in "smartDeviceLink.ini" file yet. I have not recieved any results from test team about this issue yet, so should I update this pull request with change of the comment and resolve merge conflict for CMakeLists.txt, and notify test team about this PR again?

**Comment by dcherniev** [comment]
Rebuild required

**Comment by dcherniev** [comment]
Rebuild required

**Comment by dcherniev** [comment]
Rebuild required

**Comment by dcherniev** [comment]
Rebuild required

**Comment by LuxoftAKutsan** [comment]
Not actual

**Pull request closed without merging**